scipy/stats/morestats.py

```diff
@@ -3147,16 +3147,31 @@ def median_test(*args, **kwds):
     return stat, p, grand_median, table
 
 
-def _circfuncs_common(samples, high, low):
+def _circfuncs_common(samples, high, low, nan_policy='propagate'):
+    # Ensure samples are array-like and size is not zero
     samples = np.asarray(samples)
     if samples.size == 0:
-        return np.nan, np.nan
+        return np.nan, np.nan, np.nan, None
+
+    # Recast samples as radians that range between 0 and 2 pi for use in
+    # calculating the sine and cosine
+    sin_ang = (samples - low)*2.*pi / (high - low)
+    cos_ang = (samples - low)*2.*pi / (high - low)
+
```

Comment:
There were two reasons I didn't do that.  The first was to avoid adding another keyword (axis) to the function.  The second reason was because the NaN handling ended up being a bit different for circmean and circvar/std due to the different warnings that popped up in arctan2 and hypot.

I will look into this more and see if it is sensible to implement this, since it would make things more streamlined.